app/actions/remote/channel.ts

```diff
@@ -97,6 +98,81 @@ export const fetchChannelByName = async (serverUrl: string, teamId: string, chan
     }
 };
 
+export function generateChannelNameFromDisplayName(displayName: string) {
+    let name = cleanUpUrlable(displayName);
+    if (name === '') {
+        name = generateId();
+    }
+    return name;
+}
+
+export const handleCreateChannel = async (serverUrl: string, displayName: string, purpose: string, header: string, type: ChannelType) => {
+    const database = DatabaseManager.serverDatabases[serverUrl]?.database;
+    if (!database) {
+        return {error: `${serverUrl} database not found`};
+    }
+
+    let client: Client;
+    try {
+        client = NetworkManager.getClient(serverUrl);
+    } catch (error) {
+        return {error};
+    }
+
+    try {
+        const currentUserId = await queryCurrentUserId(database);
+        const currentTeamId = await queryCurrentTeamId(database);
+        const name = generateChannelNameFromDisplayName(displayName);
+        const channel = {
+            creator_id: currentUserId,
+            team_id: currentTeamId,
+            display_name: displayName,
+            header,
+            name,
+            purpose,
+            type,
+        } as Channel;
+
+        const channelData = await client.createChannel(channel);
```

Comment:
I'm posting this as I try to deterine how to update the Tables in the db.  After creating the channel on the server via `client.createChannel(channe)` I need to update the following tables of which are done through the database operators.  Some of this data is available from the returned channel (from the server).

I believe I will need to get `client.getMyChannelMember(channelData.id)` from the server to get the channel membership. 

I'll batch the operator calls.  The batched called I should be able to swtich to the channel via #5893 which I have since merged into this PR. 

* const channelRecords: Promise<Model[]> = operator.handleChannel({channels, prepareRecordsOnly: false});
  * channels the returned channel from `client.createChannel`
* const channelInfoRecords: Promise<Model[]> = operator.handleChannelInfo({channelInfos, prepareRecordsOnly: true});
  * channelInfos will be created from the returned channel from `client.createChannel`
* const membershipRecords: Promise<Model[]> = operator.handleChannelMembership({channelMemberships: memberships, prepareRecordsOnly: true});
  * channelMemberships from `client.getMyChannelMember(channelData.id)`
* const myChannelRecords: Promise<Model[]> = operator.handleMyChannel({channels, myChannels: memberships, prepareRecordsOnly: true});
* const myChannelSettingsRecords: Promise<Model[]> = operator.handleMyChannelSettings({settings: memberships, prepareRecordsOnly: true});

Reply:
I've updated this section but still having issues will continue again starting tomorrow. `switchToChannel()` is failing on the line following line.

`const channel: ChannelModel = await member.channel.fetch();`

Once I get this working, I'll batch the jobs.